com.unity.netcode.gameobjects/Runtime/Components/Helpers/ComponentController.cs

```diff
@@ -0,0 +1,543 @@
+                {
+                    continue;
+                }
+
+                if (Components[i].Component == null)
+                {
+                    continue;
+                }
+                var objectType = Components[i].Component.GetType();
+                if (objectType == typeof(GameObject))
+                {
+                    if (!gameObjectsToScan.Contains(Components[i]))
+                    {
+                        gameObjectsToScan.Add(Components[i]);
+                    }
+                    Components.RemoveAt(i);
+                    continue;
+                }
+
+                if (!IsValidComponentType(Components[i].Component))
+                {
+                    Debug.LogWarning($"Removing {GetComponentNameFormatted(Components[i].Component)} since {Components[i].Component.GetType().Name} is not an allowed component type.");
+                    Components.RemoveAt(i);
+                    continue;
+                }
+
+                var propertyInfo = Components[i].Component.GetType().GetProperty("enabled", BindingFlags.Instance | BindingFlags.Public);
+                if (propertyInfo == null && propertyInfo.PropertyType != typeof(bool))
+                {
+                    Debug.LogWarning($"{Components[i].Component.name} does not contain a public enabled property! (Removing)");
+                    Components.RemoveAt(i);
+                }
+            }
+
+            // Second pass is to process any GameObjects added.
+            // Scan the GameObject and all of its children and add all valid components to the list.
+            foreach (var entry in gameObjectsToScan)
+            {
+                var asGameObject = entry.Component as GameObject;
+                var components = asGameObject.GetComponentsInChildren<Component>();
+                foreach (var component in components)
+                {
+                    // Ignore any NetworkBehaviour derived, NetworkObject, or NetworkManager components
+                    if (!IsValidComponentType(component))
+                    {
+                        continue;
+                    }
+
+                    var propertyInfo = component.GetType().GetProperty("enabled", BindingFlags.Instance | BindingFlags.Public);
+                    if (propertyInfo != null && propertyInfo.PropertyType == typeof(bool))
+                    {
+                        var componentEntry = new ComponentEntry()
+                        {
+                            Component = component,
+                            PropertyInfo = propertyInfo,
+                        };
+                        Components.Add(componentEntry);
+                    }
+                }
+            }
+            gameObjectsToScan.Clear();
+
+            // Final (third) pass is to name each list element item as the component is normally viewed in the inspector view.
+            for (int i = 0; i < Components.Count; i++)
```

Comment:
very nit: This might be simpler to read as a `foreach` loop. I do not have strong feelings about this.

Reply:
No worries... I can do that for sure...